app/models/apple/api.rb

```diff
@@ -0,0 +1,290 @@
+# frozen_string_literal: true
+
+require 'uri'
+require 'net/http'
+require 'base64'
+
+module Apple
+  class Api
+    ERROR_RETRIES = 3
+    SUCCESS_CODES = %w(200 201).freeze
+
+    attr_accessor :provider_id, :key_id, :key, :bridge_url
+
+    def self.from_env
+      apple_key_pem = Base64.decode64(ENV['APPLE_KEY_PEM_B64'])
+
+      new(provider_id: ENV['APPLE_PROVIDER_ID'],
+          key_id: ENV['APPLE_KEY_ID'],
+          key: apple_key_pem)
+    end
+
+    def self.from_apple_credentials(apple_credentials)
+      if apple_credentials.no_apple_credentials?
+        Rails.logger.info("No Apple credentials found via creds #{apple_credentials.id}, falling back to environment")
+        from_env
+      else
+        new(provider_id: apple_credentials.apple_provider_id,
+            key_id: apple_credentials.apple_key_id,
+            key: apple_credentials.apple_key)
+      end
+    end
+
+    def initialize(provider_id:, key_id:, key:, bridge_url: nil)
+      bridge_url = ENV.fetch('APPLE_API_BRIDGE_URL') unless bridge_url.present?
+
+      @provider_id = provider_id
+      @key_id = key_id
+      @key = key
+      @bridge_url = URI(bridge_url)
+    end
+
+    def ec_key
+      @ec_key ||= OpenSSL::PKey::EC.new(key)
+    end
+
+    def jwt_payload
+      now = Time.now.utc
+
+      { iss: provider_id,
+        exp: now.to_i + (60 * 15),
+        aud: 'podcastsconnect-v1' }
+    end
+
+    def jwt_headers
+      { kid: key_id, typ: 'JWT' }
+    end
+
+    def jwt
+      JWT.encode(jwt_payload, ec_key, 'ES256', jwt_headers)
+    end
+
+    def api_base
+      'https://api.podcastsconnect.apple.com/v1/'
+    end
+
+    def list_shows
+      get('shows')
+    end
+
+    def set_headers(req)
+      req['Authorization'] = "Bearer #{jwt}"
+      req['Content-Type'] = 'application/json'
+      req['User-Agent'] = "PRX-Feeder-Apple/1.0 (Rails-#{Rails.env})"
```

Comment:
Oh interesting - I guess I thought the api-bridge-lambda would GET the enclosure from DTR (the redirect), download from the dovetail CDN, then upload it to Apple.  But I guess it's Feeder (rails) doing the downloading?

In any case, it's a good idea to set a user-agent.